server/src/app.rs

```diff
@@ -0,0 +1,52 @@
+use actix_web::{get, post, web, Error, HttpResponse};
+use config::Config;
+use crate::graphql::{make_schema, Context, Schema};
+use juniper::http::{graphiql::graphiql_source, GraphQLRequest};
+use std::sync::Arc;
+
+/// State store shared across request invocations. Only available to requests
+/// as an immutable reference, so use interior mutability for mutable state
+pub struct AppData {
+    schema: Schema,
+    context: Context,
+    config: Config
+}
+
+impl AppData {
+    pub fn new(config: Config) -> AppData {
+        AppData {
+            schema: make_schema(),
+            context: Context::new(),
+            config
+        }
+    }
+}
+
+/// Handler to execute a GraphQL request (either a query or a mutation)
+#[post("/graphql")]
+async fn graphql(
+    data: web::Data<Arc<AppData>>,
+    req: web::Json<GraphQLRequest>,
+) -> Result<HttpResponse, Error> {
```

Comment:
is this autoformatted? the indentation and stuff looks a little weird. Also lets get rid of the web::Json<GraphQLRequest>?

Reply:
Yeah it is :-\

The web::Json is a wrapper that abstracts away deserialization of the request - without it we'd have to implement deserializing of a GraphQLRequest from the request parameters. See https://docs.rs/actix-web/2.0.0/actix_web/web/struct.Json.html

I'm open to consider eliminating JSON at some point in the future but for the sake of simplicity I think we should keep using the convenience wrapper for now.